### PR TITLE
Issue 49 and 50

### PR DESCRIPTION
**Pull request by jdr0887**
The queryservice integration works & the output seems legit.

### CODE REVIEW

**Comment by jdr0887** [comment]
After reviewing the python, there is still more to do...however, not sure if it applicable since it involves "provenance" & "extra_(nodes/edges)".  Do we want to incorporate this or folllow the TranslatorAPI strictly?

**Comment by balhoff** [comment]
> After reviewing the python, there is still more to do...however, not sure if it applicable since it involves "provenance" & "extra_(nodes/edges)". Do we want to incorporate this or folllow the TranslatorAPI strictly?

These correspond to my 3 and 4 in this ticket: #35. I think we should first skip those, deploy a working app, then implement as query options to start. We will end up needing to take this to the API group for discussion, but it will be useful to have a working version to point at.

File: src/main/scala/org/renci/cam/QueryService.scala [hunk 1]
```diff
@@ -181,11 +189,83 @@ object QueryService extends LazyLogging {
...
+    } yield TranslatorMessage(Some(queryGraph), Some(TranslatorKnowledgeGraph(kgNodes.toList, kgEdges.toList)), Some(results))
+
+  def getProvenance(prefixes: String, source: String, predicate: String, target: String): RIO[Config[AppConfig], String] =
+    for {
+      query <- Task.effect(QueryFactory.create(s"""$prefixes
+SELECT ?g ?other WHERE { GRAPH ?g { <$source> <$predicate> <$target> } OPTIONAL { ?g prov:wasDerivedFrom ?other . }
+}"""))
+      bindings <- SPARQLQueryExecutor.runSelectQuery(query)
+      nextSolution = bindings.nextSolution()
+      prov <- Task.effect(nextSolution.get("g").toString).orElse(Task.effect(nextSolution.get("other").toString))
```
Comment (balhoff) [inline src/main/scala/org/renci/cam/QueryService.scala:253]: I think this is backwards. If `?other` is bound, then that means the triple was found in an inferred graph, which was derived from the graph in `?other`. So take `?other` first, if it is not bound then take `?g` (meaning the triple is in the asserted graph).

File: src/main/scala/org/renci/cam/domain/package.scala [hunk 2]
```diff
@@ -2,14 +2,28 @@ package org.renci.cam
...
+  case class TranslatorQueryEdge(id: String, `type`: String, source_id: String, target_id: String)
 
-  case class KGSQueryGraph(nodes: List[KGSNode], edges: List[KGSEdge])
+  case class TranslatorQueryGraph(nodes: List[TranslatorQueryNode], edges: List[TranslatorQueryEdge])
 
-  case class KGSNode(id: String, `type`: String, curie: Option[String])
+  case class TranslatorNodeAttribute(name: String, value: String, `type`: String, url: Option[String], source: Option[String])
 
-  case class KGSEdge(id: String, source_id: String, target_id: String, `type`: String)
+  case class TranslatorNode(id: String, name: Option[String], `type`: Option[List[String]], attributes: Option[List[TranslatorNodeAttribute]])
```
Comment (balhoff) [inline src/main/scala/org/renci/cam/domain/package.scala:13]: Do you think we need to nest these `List`s in `Option`? Or just have an empty list if none?

Comment (balhoff) [reply src/main/scala/org/renci/cam/domain/package.scala:13]: Unexpectedly, removing the Option solves the stack overflow problem. Can we do that?

Comment (balhoff) [reply src/main/scala/org/renci/cam/domain/package.scala:13]: Some info here: https://circe.github.io/circe/codecs/known-issues.html

Comment (jdr0887) [reply src/main/scala/org/renci/cam/domain/package.scala:13]: I was assuming that anything not marked required in the api should be wrapped in an Option. For lists, I think that assumption is wrong & just having empty lists makes sense.

File: src/main/scala/org/renci/cam/QueryService.scala (hunk 1 above)
Comment (jdr0887) [reply src/main/scala/org/renci/cam/QueryService.scala:258]: Easy enough to fix.

File: src/main/scala/org/renci/cam/domain/package.scala (hunk 2 above)
Comment (jdr0887) [reply src/main/scala/org/renci/cam/domain/package.scala]: Nice find on the known-issues.  I was having a terrible time finding the root cause as Scala, SBT, & Intellij are relatively new to me.

**Pull request merged**